python/tank/deploy/tank_commands/validate_config.py

```diff
@@ -113,30 +246,139 @@ def _run(self, log):
         log.info("")
         log.info("")
         log.info("")
-        
-        
 
+    def _dump(self, log, envs, params):
+        """
+        Dump the supplied environments with the specified parameters
+
+        :param log: A logger instance.
+        :param envs: list of environment objects
+        :param params: parameter dict.
+        """
+
+        if params["dump_sparse"]:
+            dump_type = environment.UpdateAllSettingsFormat.SPARSE
+            dump_display_type = "sparse"
+        else:
+            dump_type = environment.UpdateAllSettingsFormat.FULL
+            dump_display_type = "full"
+
+        for env in envs:
+
+            # build the full output path from the dump location directory
+            # and the basename of the source environment file.
+            output_path = os.path.abspath(
+                os.path.join(params["dump_location"],
+                os.path.basename(env.disk_location))
+            )
+
+            # make sure the output path doesn't match the environment path.
+            # don't allow writing over existing environments.
+            if output_path == env.disk_location:
+                raise TankError(
+                    "Dump locaiton matches the existing environment path. "
+                    "Refusing to overwrite existing environment file."
+                )
+
+            # build a copy of the environment file that we can use to process.
+            (filename, ext) = os.path.splitext(
+                os.path.basename(env.disk_location)
+            )
+            tmp_env_path = os.path.join(
```

Comment:
feels like we are jumping through lots of hoops here to work around the fact that environment writes to a particular file. Instead could it use something like https://docs.python.org/2/library/stringio.html and we could do something like:

```python
# make a file-like object that the environment will write into
env_file = StringIO.StringIO()
# get env via official method on pc
writable_env = self.tk.pipeline_configuration.get_environment(env_name, writable=True)
# new method on writable_environment to make it easy to write the environment out 
# to an alternative location.
new_env = writable_env.clone(env_file)
# sparsify our new env
new_env.make_sparse()
# get string content of new config
env_content = env_file. getvalue()
```